server/src/main/java/com/thoughtworks/go/config/ConfigConverter.java

```diff
@@ -639,4 +641,395 @@ private TrackingTool toTrackingTool(CRTrackingTool crTrackingTool) {
+        return new CRPackageMaterial(packageMaterialConfig.getName().toString(), packageMaterialConfig.getPackageId());
+    }
+
+    public CRDependencyMaterial dependencyMaterialConfigToCRDependencyMaterial(DependencyMaterialConfig dependencyMaterialConfig) {
+        CRDependencyMaterial crDependencyMaterial = new CRDependencyMaterial(
+                dependencyMaterialConfig.getPipelineName().toString(),
+                dependencyMaterialConfig.getStageName().toString());
+        if (dependencyMaterialConfig.getName() != null)
+            crDependencyMaterial.setName(dependencyMaterialConfig.getName().toString());
+        return crDependencyMaterial;
+    }
+
+    private CRScmMaterial scmMaterialToCRScmMaterial(ScmMaterialConfig scmMaterialConfig) {
+        String materialName = null;
+        if (scmMaterialConfig.getName() != null) {
+            materialName = scmMaterialConfig.getName().toString();
+        }
+        if (scmMaterialConfig instanceof GitMaterialConfig) {
+            GitMaterialConfig git = (GitMaterialConfig) scmMaterialConfig;
+            return new CRGitMaterial(materialName, git.getFolder(), git.isAutoUpdate(), git.isShallowClone(), git.getUrl(), git.getBranch(), git.isInvertFilter(), git.getFilterAsString());
+        } else if (scmMaterialConfig instanceof HgMaterialConfig) {
+            HgMaterialConfig hg = (HgMaterialConfig) scmMaterialConfig;
+            return new CRHgMaterial(materialName, hg.getFolder(), hg.isAutoUpdate(), hg.getUrl(), hg.isInvertFilter(), hg.getFilterAsString());
+        } else if (scmMaterialConfig instanceof P4MaterialConfig) {
+            P4MaterialConfig p4MaterialConfig = (P4MaterialConfig) scmMaterialConfig;
+
+
+            CRP4Material crP4Material = new CRP4Material(materialName, p4MaterialConfig.getFolder(), p4MaterialConfig.isAutoUpdate(), p4MaterialConfig.getServerAndPort(), p4MaterialConfig.getView(), p4MaterialConfig.getUserName(), null, p4MaterialConfig.getUseTickets(), p4MaterialConfig.isInvertFilter(), p4MaterialConfig.getFilterAsString());
+
+            if (p4MaterialConfig.getEncryptedPassword() != null) {
+                crP4Material.setEncryptedPassword(p4MaterialConfig.getEncryptedPassword());
+            }
+
+            return crP4Material;
+        } else if (scmMaterialConfig instanceof SvnMaterialConfig) {
+            SvnMaterialConfig svnMaterial = (SvnMaterialConfig) scmMaterialConfig;
+            CRSvnMaterial crSvnMaterial = new CRSvnMaterial(materialName, svnMaterial.getFolder(), svnMaterial.isAutoUpdate(),
+                    svnMaterial.getUrl(), svnMaterial.getUserName(), null, svnMaterial.isCheckExternals(), svnMaterial.isInvertFilter(), svnMaterial.getFilterAsString());
+            if (svnMaterial.getEncryptedPassword() != null) {
+                crSvnMaterial.setEncryptedPassword(svnMaterial.getEncryptedPassword());
+            } else {
+                crSvnMaterial.setPassword(svnMaterial.getPassword());
```

Comment:
Shouldn't we just use encrypted password here as well? The `SvnMaterialConfig#ensureEncrypted` method should ensure that the password is encrypted, when it is in this form. Right?

Reply:
I'm unsure. I did it this way because users are able to specify a plain text password? @tomzo thoughts?

Reply:
Yes users can have plain text passwords, if they want to. This applies to svn, perforce and TFS.
But what @arvindsv  had in mind was probably that now we can enforce encrypted passwords on user - by default exported pipeline would have encrypted password.

Reply:
So, @arvindsv should I be using the encrypted password for all of the materials?

Reply:
@ibnc Yes, we should. No reason to export with a plain-text password. What @tomzo said. Thank you!

Reply:
Just to be clear, I believe it's already done for other materials.

Reply:
👍